save-deploy/base-images/Dockerfile

```diff
@@ -14,9 +14,13 @@ RUN if [ "$BASE_IMAGE_NAME" = "python" ]; then \
     fi
 
 RUN groupadd --gid 1100 save-agent && \
+    groupadd --gid 1200 save-executor && \
     useradd --uid 1100 --gid 1100 --create-home --shell /bin/sh save-agent && \
+    useradd --uid 1200 --gid 1100 --create-home --shell /bin/sh save-agent && \
+    usermod -aG save-executor save-agent && \
```

Comment:
I cannot understand why do we create a new user **and** remove the key from the pod?
As far as I can see, there are two ways: 
1. Leave the key in `/var/run/secrets/kubernetes.io/serviceaccount/token` and forbid the child process of save-agent accessing it (probably create another user as you do here);
2. Forget about Unix permission system and delegate token management to orchestrating pod.

To my mind second way brings less problems as in case of orchestrator sending token to agent might cause problems with save-demo-agent (it has ktor server running and so we might need to implement auth for ktor server also). What was your idea?

Sorry for bothering and thanks in advance

Reply:
The problem with the second approach is that you'd need to reliably detect requests from not-yet-authorized agents and distinguish them from similar requests from a tested tool. One idea (I believe, that's why I've added `Agent.isAuthenticated`) was to initialize agents with `isAuthenticated = false`, then these agents would send request for a token (which they would do strictly before starting a tested tool), orchestrator would update `isAuthenticated = true` and won't return token for any new requests from this agent. However, there is no guarantee that already running tool will be able to mimic as a newly created agent and get the token.
That's why the first approach seemed more viable: The agent would read token from a file, that has restricted access for a user, that runs a tested tool.

Reply:
Thanks! Now I got the idea